src/osx/cocoa/dataview.mm

```diff
@@ -2400,39 +2400,34 @@ void UpdateWithRow(int row)
         // previous width in that case because it must not get lost.
         nativeData->SetPrevWidth(GetColumn(pos)->GetWidth());
 
-        if ( GetColumn(pos)->GetWidthVariable() == wxCOL_WIDTH_AUTOSIZE )
-        {
-            [column setMinWidth:autoWidth];
-            [m_OutlineView sizeLastColumnToFit];
-            int minW = GetColumn(pos)->GetMinWidth();
-            [column setMinWidth:minW];
-        }
-        else
-        {
-            int minW = GetColumn(pos)->GetMinWidth();
-            if (minW == 0) {
-                [column setMinWidth:10];
-            } else {
-                [column setMinWidth:minW];
-            }
-            [column setMinWidth:minW];
-            [m_OutlineView sizeLastColumnToFit];
-            if (minW == 0) {
-                [column setMinWidth:0];
-            }
+        // Compute the remaining space for the last column; note that we
+        // deliberately don't use sizeLastColumnToFit() because it causes a
+        // refresh of the NSTableView; using sizeLastColumnToFit() would also
+        // require us to temporarily enable NSTableColumnAutoresizingMask on
+        // the NSTableColumn because it doesn't seem to work without it
+        NSSize size = [[m_OutlineView superview] bounds].size;
+        size.width -= (noOfColumns * [m_OutlineView intercellSpacing].width);
```

Comment:
This seems wrong when having one column, as there is no neighbouring column requiring spacing. However if the left and right marging are treated as cell spacing then changing this to:

```suggestion
        size.width -= ((noOfColumns + 1) * [m_OutlineView intercellSpacing].width);
```

gets rid of horizontal scrollbars with macOS 11. I can increase `size.width` by 2 more, with 3 a scrollbar appears again. I haven't been able to find out more about the margin extent.

(I'm aware this PR is closed but didn't want this info to go completely missing)